src/main/java/com/gtnewhorizon/gtnhlib/config/ConfigurationManager.java

```diff
@@ -92,36 +91,128 @@ private static void processConfigInternal(Class<?> configClass, String category,
+                                }
+                                return null;
+                            }).orElse(null);
+                    if (newDefault != null) defaultValue = Boolean.parseBoolean(newDefault);
+                }
                 field.setBoolean(null, rawConfig.getBoolean(name, category, defaultValue, comment, langKey));
             } else if ((boxed = fieldClass.equals(Integer.class)) || fieldClass.equals(int.class)) {
                 val range = Optional.ofNullable(field.getAnnotation(Config.RangeInt.class));
                 val min = range.map(Config.RangeInt::min).orElse(Integer.MIN_VALUE);
                 val max = range.map(Config.RangeInt::max).orElse(Integer.MAX_VALUE);
-                val defaultValue = Optional.ofNullable(field.getAnnotation(Config.DefaultInt.class))
+                int defaultValue = Optional.ofNullable(field.getAnnotation(Config.DefaultInt.class))
                         .map(Config.DefaultInt::value).orElse(boxed ? (Integer) field.get(null) : field.getInt(null));
+                if (field.isAnnotationPresent(Config.ModDetectedDefault.class)) {
+                    val newDefault = Optional.ofNullable(field.getAnnotation(Config.ModDetectedDefault.class))
+                            .map(f -> {
+                                if (!f.coremod().isEmpty()) {
+                                    try {
+                                        Class.forName(f.coremod());
+                                        return f.value();
+                                    } catch (ClassNotFoundException ignored) {}
+                                }
+                                if (!f.modID().isEmpty()) {
+                                    if (Loader.isModLoaded(f.modID())) {
+                                        return f.value();
+                                    }
+                                }
+                                return null;
+                            }).orElse(null);
+                    if (newDefault != null) defaultValue = Integer.parseInt(newDefault);
+                }
                 field.setInt(null, rawConfig.getInt(name, category, defaultValue, min, max, comment, langKey));
             } else if ((boxed = fieldClass.equals(Float.class)) || fieldClass.equals(float.class)) {
                 val range = Optional.ofNullable(field.getAnnotation(Config.RangeFloat.class));
                 val min = range.map(Config.RangeFloat::min).orElse(Float.MIN_VALUE);
                 val max = range.map(Config.RangeFloat::max).orElse(Float.MAX_VALUE);
-                val defaultValue = Optional.ofNullable(field.getAnnotation(Config.DefaultFloat.class))
+                float defaultValue = Optional.ofNullable(field.getAnnotation(Config.DefaultFloat.class))
                         .map(Config.DefaultFloat::value).orElse(boxed ? (Float) field.get(null) : field.getFloat(null));
+                if (field.isAnnotationPresent(Config.ModDetectedDefault.class)) {
```

Comment:
This seems repeated amongst all of the cases - would be nice to turn it into a function/class helper that could be used in the chain above...